packages/ui/Input/index.tsx

```diff
@@ -0,0 +1,190 @@
+import * as React from "react";
+import styled, { css } from "styled-components";
+
+import { Icon } from "../Icon";
+
+export const Input = ({
+  label,
+  hasError = false,
+  id,
+  supportingLabel,
+  type,
+  value,
+  onChange,
+}: InputProps) => {
+  const randomId = React.useId();
+  const [typeOverride, setTypeOverride] = React.useState("");
+  const inputRef = React.useRef<HTMLInputElement | null>(null);
+
+  const getButton = () => {
+    if (hasError) {
+      return (
+        <StyledIcon>
+          <Icon icon="error" filled color="#b3261e" iconSize={20} />
+        </StyledIcon>
+      );
+    }
+    if (type === "password") {
+      return typeOverride ? (
+        <StyledIcon
+          onClick={() => {
+            setTypeOverride("");
+            inputRef.current?.focus();
+          }}
+        >
+          <Icon
+            icon="visibility_off"
+            color="#49454f"
+            iconSize={23}
+          />
+        </StyledIcon>
+      ) : (
+        <StyledIcon
+          onClick={() => {
+            setTypeOverride("text");
+            inputRef.current?.focus();
+          }}
+        >
+          <Icon
+            icon="visibility"
+            color="#49454f"
+            iconSize={23}
+          />
+        </StyledIcon>
+      );
+    }
+    if (value) {
+      return (
+        <StyledIcon
+          onClick={() => {
+            if (onChange) {
+              onChange("");
+            }
+            inputRef.current?.focus();
+          }}
+          style={{ transform: "rotate(45deg)" }}
+        >
+          <Icon
+            icon={"add_circle"}
+            color={"#49454f"}
+            iconSize={20}
+          />
+        </StyledIcon>
+      );
+    }
+    return null;
+  };
+
+  return (
+    <Wrapper>
+      <StyledInput
+        ref={inputRef}
+        placeholder=" "
+        hasError={hasError}
+        id={id || randomId}
+        type={typeOverride || type}
+        value={value}
+        onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+          if (onChange) {
+            onChange(event.target.value);
+          }
+        }}
```

Comment:
```suggestion
onChange={onChange}
```
Someone that uses this input probably knows how he need to handle onChange. Let's just pass their function instead.🚀